### PR TITLE
Pyclient contract abstraction [depends on #184]

### PR DESCRIPTION
**Pull request by dtebbs**
Changes related to contract deployment, invocation etc.
Expose some useful utility functions in client code, on contracts and for testing.

### CODE REVIEW

File: pyClient/commands/zeth_deploy.py [hunk 1]
```diff
@@ -43,14 +42,12 @@ def deploy(
     token_instance_desc = get_erc20_instance_description(token_address) \
         if token_address else None
 
-    zeth_client = MixerClient.deploy(
+    _zeth_client, mixer_instance_desc = MixerClient.deploy(
```
Comment (AntoineRondelet) [inline pyClient/commands/zeth_deploy.py:45]: What is  the leading  `_` in `_name_of_var` supposed to represent? (put otherwise, why did you rename `zeth_client` to `_zeth_client`?)

File: pyClient/test_commands/test_ether_mixing.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def main() -> None:
 
     # Deploy Zeth contracts
     tree_depth = zeth.constants.ZETH_MERKLE_TREE_DEPTH
-    zeth_client = MixerClient.deploy(
+    zeth_client, _contract_desc = MixerClient.deploy(
```
Comment (AntoineRondelet) [inline pyClient/test_commands/test_ether_mixing.py:53]: Same question as above: Does the leading "_" have a meaning here?

File: pyClient/commands/zeth_deploy.py (hunk 1 above)
Comment (dtebbs) [reply pyClient/commands/zeth_deploy.py:45]: For variables, the leading `_` marks the variable as unused preventing the linting tools from complaining.

Comment (AntoineRondelet) [reply pyClient/commands/zeth_deploy.py:45]: Oh alright I see. Isn't it possible to replace these unused variables by a `_` then? (i.e. `_, mixer_instance_desc = MixerClient.deploy(`

Comment (dtebbs) [reply pyClient/commands/zeth_deploy.py:45]: AFAIK `_` is not treated as special, as it is in some languages, (except where it is used for "last value" in the REPL, IIRC) so it get's a value and type like any other variable. Hence it usually creates type conflicts if used multiple times per scope.  So I think it's possible, but not necessarily everywhere, and in Python it doesn't seem to have an advantage over the `_` prefixed version (which also includes information about the unused variable).  At least, that's my understanding.

Comment (AntoineRondelet) [reply pyClient/commands/zeth_deploy.py:45]: I see - it's not like in Go...
I just read that a way to get around that is to select the outputs of a function by doing something like `desired_output1, desired_output2 = function(args..)[indices_of_desired_outputs]`. Haven't tried it though...

Comment (dtebbs) [reply pyClient/commands/zeth_deploy.py:45]: Interesting. I hadn't seen that approach before.
